### PR TITLE
add no-print rule to enforce use of logger

### PR DESCRIPTION
**Pull request by slheavner**
Similar to the `no-console` rules for eslint/tslint. Enforces the use of a logging framework rather than raw print statements.

### CODE REVIEW

**Comment by slheavner** [comment]
bump @elsassph

File: src/plugins/codeStyle/index.ts [hunk 1]
```diff
@@ -56,6 +56,11 @@ export default class CodeStyle {
                         );
                     }
                 }
+            },
+            PrintStatement: s => {
+                if (noPrint) {
```
Comment (elsassph) [inline src/plugins/codeStyle/index.ts:61]: If not `off`

File: src/plugins/codeStyle/diagnosticMessages.ts [hunk 2]
```diff
@@ -77,6 +78,12 @@ const messages = {
         code: CodeStyleError.TypeAnnotation,
         message: `${ST} type annotation required`,
         range
+    }),
+    noPrint: (range: Range, severity: DiagnosticSeverity) => ({
+        severity: severity,
+        code: CodeStyleError.NoPrint,
+        message: `${CS} don't use direct Print statements`,
```
Comment (elsassph) [inline src/plugins/codeStyle/diagnosticMessages.ts:85]: While we're doing another round of review, maybe:
"Avoid using direct Print statements"?

File: src/plugins/codeStyle/index.ts (hunk 1 above)
Comment (slheavner) [reply src/plugins/codeStyle/index.ts:61]: So I made no-print rules a `RulesSeverity` and I actually meant to remove this if. 
`        'no-print'?: RuleSeverity;`
If you'd rather it be a different type lmk

File: src/plugins/codeStyle/diagnosticMessages.ts (hunk 2 above)
Comment (slheavner) [reply src/plugins/codeStyle/diagnosticMessages.ts:85]: sounds proper

**Comment by elsassph** [comment]
@slheavner I think you should remove the `package-lock.json` from your PR. This upgrades forces others to use npm 7 apparently.

**Comment by slheavner** [comment]
was wondering why that updated, reverted.

**Comment by elsassph** [comment]
Thanks!

**Pull request merged**